### PR TITLE
Free up space and clean up high scores a bit

### PR DESCRIPTION
**Pull request by yyyc514**
- Also uses enum for game state

### CODE REVIEW

**Comment by BlueMaxDX** [comment]
Thank you @Dreamer3!

File: Rogue1.4/Rogue1.4.ino [hunk 1]
```diff
@@ -20,7 +22,7 @@ struct Score {        //my first 'structure'
   byte depth;
 };
 
-struct hero {
+typedef struct Hero {
```
Comment (Pharap) [inline Rogue1.4/Rogue1.4.ino]: @yyyc514, why all the `typedef` nonsense?
This is C++, not C, you don't need to `typedef` structs.

File: Rogue1.4/HelloWorld9red7str.ino [hunk 2]
```diff
@@ -224,7 +224,14 @@ const char mes[23][10] PROGMEM = {
   "ZZZ..."            //22
 };
 */
-byte gstate = 0;
+enum GameState : uint8_t {
```
Comment (Pharap) [inline Rogue1.4/HelloWorld9red7str.ino]: You ought to be using `enum class` rather than `enum`.
Unscoped enumerations aren't type-safe.

**Comment by yyyc514** [comment]
If we're going to have a review policy we should come up with one.  It'd be far better to get feedback before my PRs were merged. ;-)  And I'd love some serious peer review in any case.  But less helpful after the fact.

**Pull request merged**